### PR TITLE
Bump vispy, cleanup deprecation warnings.

### PR DESCRIPTION
**Pull request by Carreau**
Should decrease the number of deprecation messages in CI.

### CODE REVIEW

File: setup.cfg [hunk 1]
```diff
@@ -64,7 +64,7 @@ install_requires =
     typing_extensions
     toolz>=0.10.0
     tqdm>=4.56.0
-    vispy>=0.6.4, !=0.8.0
+    vispy>=0.9
```
Comment (tlambert03) [inline setup.cfg:67]: can/should we do this before https://github.com/napari/napari/pull/3494?

Comment (Carreau) [reply setup.cfg:67]: Yeah, we should get the above first. I scanned the PR list but dod not see it. ... Converted to draft.

Comment (tlambert03) [reply setup.cfg:67]: #3494 is in if you want to take this up again?

File: napari/window.py [hunk 2]
```diff
@@ -14,6 +14,7 @@
     from ._qt import Window
 
 except ImportError:
+    raise
```
Comment (brisvag) [inline napari/window.py:17]: What was this change for?

**Review by brisvag — CHANGES_REQUESTED** [review]
Apart from the comment below, all of these changes are included or accounted for already in #3494. I think we can close this.

**Pull request closed without merging**